.github/workflows/test.yml

```diff
@@ -78,18 +121,12 @@ jobs:
           env_vars: OS, python
 
   test-integration:
+    needs: prepare-integration-test-matrix
     timeout-minutes: 20
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
-
-      matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
-        os: [ubuntu-latest, macos-latest]
-        borg-version: ["1.1.18", "1.2.2", "1.2.4", "2.0.0b5"]
-        exclude:
-          - borg-version: "2.0.0b5"
-            python-version: "3.8"
```

Comment:
Biggest hurdle I ran into with the integration tests is excluding the combination of Borg 2.0.0b5 with python 3.8. Skipped this for now.

Reply:
Doesn't adding the exclude to the matrix work?

Reply:
Unfortunately it does not work with the way I'm dynamically creating the matrix, this only works when the matrix is static I believe. However, I have included a new 'if' conditional that I believe accomplishes the same thing.